docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+
+The most common quantities are: energy, molecular-orbital coefficients,
+gradients, vibrational frequencies, and geometries (including minimum energy,
+transition states, reaction paths, etc.).
+
+Limited inputs
+--------------
+Generally speaking the inputs required for the aforementioned results tend to be
+the molecule, level of theory (method plus basis set), and method parameters.
+While the set of potential parameters is huge, again there are only a handful of
+commonly used parameters.
+
+Primary targets are: algorithm variations (e.g. density fitting), number of
+iterations, and convergence criteria.
+
+Common Quantum Chemistry Calculations
+=======================================
+As described in the design principles, the UI should enable users to conduct
+commonly used quantum chemistry calculations such as single-point energies and
+geometry optimizations in as few steps as possible. It should also be flexible
+enough to allow for more complex workflows using the helper/driver functions and
+PluginPlay modules. Users can execute these calculations either via Python
+scripts or interactively using Jupyter notebooks. For those who desire a
+fine-grained control of their workflow, they can utilize the Python and/or C++
+API.
+
+1. **Single Point Energy**
+   
+   In these calculations total energy is computed for a fixed geometry of the
+   molecule (arrangement of atomic nuclei) corresponding to a single point in
+   the potential energy surface. While energy calculation is the most common
+   case, the user might be interested in obtaining the wave function,
+   derivatives, or other properties (dipole moment, electron density, etc.) for
+   this specific geometry. 
+
+2. **Geometry Optimization**
+   
+   This is a procedure to find the arrangement of atomic nuclei that corresponds
+   to a stationary point in the potential energy surface. This procedure
+   generally requires the calculation of the gradients and Hessian (first and
+   second order partial derivatives of the energy with respect to nuclear
+   coordinates, respectively) at many different geometries. 
+
+3. **Vibrational Frequency Calculation**
+   
+   These calculations provide the vibrational modes of a molecule and their
+   corresponding frequencies. The procedure requires Hessian calculation at the
+   equilibrium  geometry.
+
+4. **Molecular Dynamics Simulation**
+   
+   These calculations allow for the study of the trajectory of atoms and
+   molecules over time using classical dynamics. Gradient calculations are
+   required to compute the forces acting on the atoms.
+
+We will initially focus on the UX design of the "single point energy"
+calculations in this document since other types of calculations (geometry
+optimization, vibrational frequency calculation, etc.) can be performed through
+external codes rather easily as long as the UX for "single point calculations"
+provides the appropriate support. 
```

Comment:
```suggestion
optimization, vibrational frequency calculation, etc.) require similar inputs.
```
I agree we don't need to worry about them ont this page, but I disagree with the reason. I'd argue that performing geometry optimization/vibrational frequency calculations strictly through a single point UI, would mean that the user would not have access to analytic derivatives. That's going to be a huge performance hit.

Reply:
I second this point.